app/services/storage_location_deactivate_service.rb

```diff
@@ -0,0 +1,21 @@
+class StorageLocationDeactivateService
+  include ServiceObjectErrorsMixin
+
+  def initialize(storage_location)
+    @storage_location = storage_location
+  end
+
+  def call
+    raise Errors::StorageLocationNotEmpty unless valid?
```

Comment:
Instead of raising an error could we return false and include an error message? It is one less error class that we need to manage and it is only specific to this location.